File: core/runtime/runtime_recovery_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.runtime.runtime_state_machine_engine import RuntimeStateMachine


def recover_runtime(failed_state: str, evidence: list | None = None) -> Dict[str, Any]:
    sm = RuntimeStateMachine()
    if failed_state != "initialized":
        sm.transition(failed_state, evidence=evidence)
    sm.transition("retrying", evidence=evidence or ["recovery"])
    sm.transition("running", evidence=evidence or ["recovery"])
    return {
        "recovered_state": sm.state,
        "transitions": len(sm.history),
        "deterministic": True,
    }

File: src/runtime/runtimeRecoveryEngine.ts
/**
 * Converted from Python: core/runtime/runtime_recovery_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "./pyCompat.js";
import { RuntimeStateMachine } from "./runtimeStateMachineEngine.js";

export function recoverRuntime(failed_state: any, evidence: any = null): any {
  var sm: any = new RuntimeStateMachine();
  if (!py.eq(failed_state, "initialized")) {
    sm.transition(failed_state, evidence);
  }
  sm.transition("retrying", py.or2(evidence, () => (["recovery"])));
  sm.transition("running", py.or2(evidence, () => (["recovery"])));
  return {"recovered_state": sm.state, "transitions": py.len(sm.history), "deterministic": true};
}
export { RuntimeStateMachine };
